src/identity/interaction/email-password/credentials/ClientCredentialsAdapterFactory.ts

```diff
@@ -0,0 +1,56 @@
+import type { AdapterPayload, Adapter } from 'oidc-provider';
+import type { KeyValueStorage } from '../../../../storage/keyvalue/KeyValueStorage';
+import type { AdapterFactory } from '../../../storage/AdapterFactory';
+import { PassthroughAdapterFactory, PassthroughAdapter } from '../../../storage/PassthroughAdapterFactory';
+
+export interface ClientCredentials {
+  secret: string;
+  webId: string;
+}
+
+/**
+ * A {@link PassthroughAdapter} that overrides the `find` function
+ * by checking if there are stored client credentials for the given ID
+ * if no payload is found in the source.
```

Comment:
I see how it works, but not why.

Can you explain what this class is and what it is used for?
The current comment text might fit better on the `find` method.

Reply:
The comment does start from the assumption that the reader knows what an `Adapter` is. I can't add comments to the `Adapter` interface since that is external, but I could add more to the `AdapterFactory` interface.